### ISSUE TITLE
AssertionError: idx 0 != idx_new 20 during testing.

### ISSUE THREAD

**Issue by pmgautam** [FangShancheng/ABINet#92]
I'm getting the error `AssertionError: idx 0 != idx_new 20 during testing.`. Can someone help me with this. I've tried looking in issues but couldn't get much help. Thanks in advance!

**Comment by fuzheng1209** [comment]
I have the same error,do you solve the problem?

**Comment by neverstoplearn** [comment]
I meet tehe save error,have anyone solve it?

**Comment by shivpoojansaini** [comment]
> I'm getting the error `AssertionError: idx 0 != idx_new 20 during testing.`. Can someone help me with this. I've tried looking in issues but couldn't get much help. Thanks in advance!

have you got this while training the model? Or only inferencing on your own data.

**Comment by shivpoojansaini** [comment]
> I meet tehe save error,have anyone solve it?

One possible solution https://github.com/FangShancheng/ABINet/issues/34#issuecomment-1465623885

**Comment by neverstoplearn** [comment]
> > I meet tehe save error,have anyone solve it?
> 
> One possible solution [#34 (comment)](https://github.com/FangShancheng/ABINet/issues/34#issuecomment-1465623885)

will this affect the performance of the model？

**Comment by shivpoojansaini** [comment]
> > > I meet tehe save error,have anyone solve it?
> > 
> > 
> > One possible solution [#34 (comment)](https://github.com/FangShancheng/ABINet/issues/34#issuecomment-1465623885)
> 
> will this affect the performance of the model？

No, we just skipped some data there is no modification from the model perspective.

**Comment by shivpoojansaini** [comment]
Fo more details understand the exception cause of `def get(self, idx):` under dataset.py

**Comment by fuzheng1209** [comment]
I use Baidu cloud datasets provided by the author.But t also meet the same time.

**Comment by fuzheng1209** [comment]
I use Baidu cloud datasets provided by the author.But I also meet the same problem.

**Comment by shivpoojansaini** [comment]
Modify the function under dataset.py 
`  if not self.is_training:
            if idx!=idx_new:
                pass`

under function `def __getitem__(self, idx):` 
Since ` def get(self, idx):` going in exception for some id of data
